src/api/pickAppResource.ts

```diff
@@ -12,10 +12,6 @@ export async function pickAppResource<T extends AzExtTreeItem>(context: ITreeIte
     const subscription: SubscriptionTreeItem = await ext.appResourceTree.showTreeItemPicker(SubscriptionTreeItem.contextValue, context);
     const appResource = await subscription.pickAppResource(context, options);
 
-    if (!appResource.resolveResult) {
```

Comment:
I moved this down to `SubscriptionTreeItem` just so that both varieties of `resolve` (one for resolving all ahead of time if needed, the other for resolving the picked item) happen in the same method body.